### ISSUE TITLE
Calendar component in DataTable PrimeReact 10.x: Date selection sets value to null, Clear button works as expected

### ISSUE THREAD

**Issue by darioEscuderop** [primefaces/primereact#7389]
### Describe the bug

There is an issue with the Calendar component in PrimeReact version 10.x. When selecting a date by clicking on it, the value of the calendar remains null, and the date is not set in the input field. This behavior occurs despite the "Clear" button working as expected, correctly clearing the value and triggering focus lose.

In version 9.5.0, this issue does not occur, and both the "Clear" button and date selection work as expected. However, in version 10.x, selecting a date via click results in a null value.

In 9.5.0 after clicking the "Clear" button in the calendar, the focus remains on the input field, which allows the user to click "Clear" multiple times without closing the calendar or losing focus. But this is less important than what happend in 10.X version.

### Reproducer

https://stackblitz.com/edit/vitejs-vite-5wzdws?file=src%2FApp.jsx

### System Information

```Shell
npmPackages:
    primereact: ^10.8.3 => 10.8.4 
    react: ^18.2.0 => 18.2.0
```

### Steps to reproduce the behavior

1. Open the calendar and click on a date to select it.
2. Observe that after selecting the date, the value remains null in the input field.

### Expected behavior

After selecting a date by clicking on it in the calendar, the input field should display the selected date, and the value should be updated accordingly.

**Comment by melloware** [comment]
cc @KumJungMin can you look at this one?

**Comment by KumJungMin** [comment]
> cc @KumJungMin can you look at this one?

---

Yes, I’ll check it :)

**Comment by KumJungMin** [comment]
@darioEscuderop 

is the implementation you are expecting similar to the example shown in the video below? :)


https://github.com/user-attachments/assets/acc14294-d400-4ce1-bf61-3635bb57c383

**Comment by darioEscuderop** [comment]
@KumJungMin 

Yes, that’s exactly what I’m expecting. Additionally, as indicated in my code, I’d like it to always save a fixed time—specifically, 12:00 in this case. I’m not sure if that would be an issue as well. The video looks great!

Thank you!
